core/src/data/value/binary_op/decimal.rs

```diff
@@ -0,0 +1,257 @@
+use {
+    super::TryBinaryOperator,
+    crate::{data::ValueError, prelude::Value, result::Result},
+    rust_decimal::prelude::Decimal as Dec,
+    std::cmp::Ordering,
+    Value::*,
+};
+
+impl PartialEq<Value> for Dec {
+    fn eq(&self, other: &Value) -> bool {
+        match other {
+            I8(other) => *self == Dec::from(*other),
+            I64(other) => *self == Dec::from(*other),
+            F64(other) => *self == Dec::from_f64_retain(*other).unwrap(),
```

Comment:
This feature looks great. Get good knowledge thanks! 
<https://docs.rs/rust_decimal/latest/rust_decimal/prelude/struct.Decimal.html#method.from_f64_retain>

But in the PartialEq implementation, it doesn't seem like it should cause a failure.
If none is displayed, it should be treated as false. What do you think?

Reply:
good idea..  I've implemented it in the latest push.